presto-main/src/test/java/com/facebook/presto/metadata/TestInformationSchemaMetadata.java

```diff
@@ -0,0 +1,219 @@
+import com.facebook.presto.transaction.TransactionManager;
+import com.facebook.presto.type.TypeRegistry;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import io.airlift.json.JsonCodec;
+import io.airlift.slice.Slices;
+import org.testng.annotations.Test;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import static com.facebook.presto.connector.ConnectorId.createInformationSchemaConnectorId;
+import static com.facebook.presto.connector.ConnectorId.createSystemTablesConnectorId;
+import static com.facebook.presto.spi.type.VarcharType.VARCHAR;
+import static com.facebook.presto.testing.TestingSession.testSessionBuilder;
+import static com.facebook.presto.transaction.TransactionManager.createTestTransactionManager;
+import static com.google.common.collect.ImmutableSet.toImmutableSet;
+import static java.util.Arrays.stream;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+public class TestInformationSchemaMetadata
+{
+    /**
+     * Tests information schema predicate pushdown when both schema and table name are specified.
+     */
+    @Test
+    public void testInformationSchemaPredicatePushdown()
+    {
+        CatalogManager catalogManager = new CatalogManager();
+        catalogManager.registerCatalog(createTestCatalog("test_catalog", new TestConnectorMetadata()));
+        TransactionManager transactionManager = createTestTransactionManager(catalogManager);
+        Metadata metadata = new MetadataManager(
+                new FeaturesConfig(),
+                new TypeRegistry(),
+                new BlockEncodingManager(new TypeRegistry()),
+                new SessionPropertyManager(),
+                new SchemaPropertyManager(),
+                new TablePropertyManager(),
+                new ColumnPropertyManager(),
+                transactionManager);
+        TransactionId transactionId = transactionManager.beginTransaction(false);
+        Session session = testSessionBuilder()
+                .setCatalog("test_catalog")
+                .setSchema("information_schema")
+                .setClientCapabilities(stream(ClientCapabilities.values())
+                        .map(ClientCapabilities::toString)
+                        .collect(toImmutableSet()))
+                .setTransactionId(transactionId)
+                .build();
+
+        ImmutableMap.Builder<ColumnHandle, Domain> domains = new ImmutableMap.Builder<>();
+        domains.put(new InformationSchemaColumnHandle("table_schema"), Domain.singleValue(VARCHAR, Slices.utf8Slice("test_schema")));
+        domains.put(new InformationSchemaColumnHandle("table_name"), Domain.singleValue(VARCHAR, Slices.utf8Slice("test_view")));
+        Constraint<ColumnHandle> constraint = new Constraint<>(TupleDomain.withColumnDomains(domains.build()));
```

Comment:
You could also add a test where `Constraint#summary` is equal to `TupleDomain.all()` and `com.facebook.presto.spi.Constraint#predicate` is set to something that selects a proper view.